app/screens/select_server/select_server.js

```diff
@@ -68,7 +68,7 @@ export default class SelectServer extends PureComponent {
     };
 
     static getDerivedStateFromProps(props, state) {
-        if (props.serverUrl && !state.url) {
+        if (props.serverUrl && state.url === null) {
```

Comment:
Do we even need `getDerivedStateFromProps` at all? The constructor sets `this.state.url` from `this.props.serverUrl`, so this would only be needed if `this.props.serverUrl` is expected to change

Reply:
removed